### PR TITLE
MoghesEats (Adds 12 new food dishes to the game of varying origins)

### PR DESCRIPTION
**Pull request by tomixcomics**
-Adds 12 new food dishes to the game (4 Unathi dishes, 2 Galatean, 1 Tretian, 5 misc.) including some requested dishes such as unathi desserts and fettucini alfredo. Expands the use of Nakarka
-fixes bug where konaqu wasn't displaying right in the codex
-fixes phrasing on nakarka cheese wheels to be more in line with how Vaucra talk
-fixes incorrectly named muffin types

### CODE REVIEW

**Comment by tomixcomics** [comment]
!request review

File: code/modules/reagents/reagent_containers/food/snacks/burger.dm [hunk 1]
```diff
@@ -229,3 +235,15 @@
 	reagents_to_add = list(/singleton/reagent/nutriment = 4, /singleton/reagent/nutriment/protein = 4)
 	reagent_data = list(/singleton/reagent/nutriment = list("barbecue sauce" = 5, "bun" = 5), /singleton/reagent/nutriment/protein = list("ground beef" = 5))
 	bitesize = 2
+
+/obj/item/reagent_containers/food/snacks/burger/nakarka_hamburger
+	name = "nakarka hamburger" //wrote "hamburger" in full to set it apart from "nakarkaburger" which is a kois dish
+	desc = "A delicious hamburger with green Nakarka cheese. Grilling the cheese makes it palettable for Humans and Tajara as well. This meal is not to be confused with the K'ois version of the same meal. No. Seriously. DO NOT CONFUSE IT."
```
Comment (listerla) [inline code/modules/reagents/reagent_containers/food/snacks/burger.dm:241]:
```suggestion
	desc = "A delicious hamburger with green Nakarka cheese. Grilling the cheese makes it palatable for Humans and Tajara as well. This meal is not to be confused with the K'ois version of the same meal. No. Seriously. DO NOT CONFUSE IT."
```

**Review by DreamySkrell — APPROVED** [review]

File: code/modules/reagents/reagent_containers/food/snacks/cultural/unathi.dm [hunk 2]
```diff
@@ -238,3 +239,69 @@
...
+	icon = 'icons/obj/item/reagent_containers/food/cultural/unathi.dmi'
+	icon_state = "steakbites"
+	trash = /obj/item/trash/bowl_brown
+	reagents_to_add = list(/singleton/reagent/nutriment/protein = 6, /singleton/reagent/drink/caramel = 5)
+	reagent_data = list(/singleton/reagent/nutriment/protein = list("meat" = 5))
+	bitesize = 2
+	filling_color = "#3b1b06"
+
+/obj/item/reagent_containers/food/snacks/caramelized_steak_bites/update_icon()
+	var/percent_steakbites = round((reagents.total_volume / 11) * 100)
```
Comment (FluffyGhoster) [inline code/modules/reagents/reagent_containers/food/snacks/cultural/unathi.dm:255]: The percentage have to be calculated based on the total amount of reagents this takes, not against a fixed number that might not be updated if the reagents change

File: code/modules/reagents/reagent_containers/food/snacks/cultural/unathi.dm [hunk 3]
```diff
@@ -238,3 +239,69 @@
...
+	desc = "A personal pie filled with raw egg. In Unathi cultures, this is considered a dessert. Other races may consider it a high-protein main course. It is sometimes nicknamed 'wasteland pie' due to the way the top of the dough cracks as it is baked."
+	icon = 'icons/obj/item/reagent_containers/food/cultural/unathi.dmi'
+	icon_state = "egg_pie"
+	reagents_to_add = list(/singleton/reagent/nutriment/protein = 6, /singleton/reagent/nutriment = 3)
+	bitesize = 3
+	filling_color = "#e6a823"
+	reagent_data = list(/singleton/reagent/nutriment = list ("dough" = 5), /singleton/reagent/nutriment/protein = list("egg" = 5))
+
+/obj/item/reagent_containers/food/snacks/egg_pie/update_icon()
+	var/percent_eggpie = round((reagents.total_volume / 9) * 100)
```
Comment (FluffyGhoster) [inline code/modules/reagents/reagent_containers/food/snacks/cultural/unathi.dm:284]: The percentage have to be calculated based on the total amount of reagents this takes, not against a fixed number that might not be updated if the reagents change

File: code/modules/reagents/reagent_containers/food/snacks/cultural/unathi.dm [hunk 4]
```diff
@@ -238,3 +239,69 @@
...
+	desc = "A simple Ouerean recipe for serving grilled fish in an egg glaze with Ouerean spices. Beloved by all classes. You could theoretically prepare most fish like this but you have to use Orszi and Ouerean spices for that full authentic taste."
+	icon = 'icons/obj/item/reagent_containers/food/cultural/unathi.dmi'
+	icon_state = "orszi"
+	trash = /obj/item/trash/plate
+	filling_color = "#e2a191"
+	reagents_to_add = list(/singleton/reagent/nutriment/protein/seafood = 2)
+	reagent_data = list(/singleton/reagent/nutriment/protein/seafood = list("grilled fish" = 5, "spices" = 3))
+
+/obj/item/reagent_containers/food/snacks/orszi/update_icon()
+	var/percent_orszi = round((reagents.total_volume / 10) * 100)
```
Comment (FluffyGhoster) [inline code/modules/reagents/reagent_containers/food/snacks/cultural/unathi.dm:302]: The percentage have to be calculated based on the total amount of reagents this takes, not against a fixed number that might not be updated if the reagents change

File: code/modules/reagents/reagent_containers/food/snacks/noodles.dm [hunk 5]
```diff
@@ -170,3 +170,29 @@
...
+	icon = 'icons/obj/item/reagent_containers/food/noodles.dmi'
+	icon_state = "alfredo"
+	trash = /obj/item/trash/plate
+	filling_color = "#f8e8b1"
+	center_of_mass = list("x"=16, "y"=10)
+	reagents_to_add = list(/singleton/reagent/nutriment = 6, /singleton/reagent/nutriment/protein/cheese = 2)
+	reagent_data = list(/singleton/reagent/nutriment = list("pasta" = 5, "garlic" = 3), /singleton/reagent/nutriment/protein/cheese = list("creamy parmesan" = 5))
+
+/obj/item/reagent_containers/food/snacks/fettuccine_alfredo/update_icon()
+	var/percent_fettuccine_alfredo = round((reagents.total_volume / 8) * 100)
```
Comment (FluffyGhoster) [inline code/modules/reagents/reagent_containers/food/snacks/noodles.dm:193]: The percentage have to be calculated based on the total amount of reagents this takes, not against a fixed number that might not be updated if the reagents change

**Review by FluffyGhoster — CHANGES_REQUESTED** [review]

File: code/modules/reagents/reagent_containers/food/snacks/noodles.dm (hunk 5 above)
Comment (tomixcomics) [reply code/modules/reagents/reagent_containers/food/snacks/noodles.dm:193]: In all of these the number chosen is based on the total amount of reagents in the dish. It's the way it's coded in every other existing dish in the game that changes appearance based on how much of it is eaten dating back to the steak. I tested it works, the only thing that might throw it out of whack is if someone adds a significant amount of reagents to the dish such as pouring large amounts of liquid onto it and if there is a way to account for that in code i don't really know what that would be so I formatted it based on existing dishes while changing the number total to the number of reagents in a dish as it is when the dish is dcreated.

Comment (FluffyGhoster) [reply code/modules/reagents/reagent_containers/food/snacks/noodles.dm:193]: I don't see it done anywhere else like this by a quick look

You can calculate the total amount of reagents in the dish by looping the reagents_to_add list and adding up the values, or by `.maximum_volume`, depending on what you want to do exactly

Comment (tomixcomics) [reply code/modules/reagents/reagent_containers/food/snacks/noodles.dm:193]: calculating the total amount of reagents in the dish by adding the reagents_to_add list is exactly what i did and how i got the numbers i put in. Unless i am mistaken they should all have the number of reagents in them as the divisable number.

also if you are looking for other dishes that use this code there is steak, xenosteak, neaera brain loaf, packed rice bowl, as well as other dishes i created previously such as lasgna slices, floating island, north 60 squid, meatballs and peas, schnitzel, and others.

i'm not sure i understand what the change you are suggesting is. would "var/percent = round((reagents.total_volume / volume) * 100)" work better? (I don't know how to write brand new code but that's how the code for toothpaste looks)

Comment (FluffyGhoster) [reply code/modules/reagents/reagent_containers/food/snacks/noodles.dm:193]: >calculating the total amount of reagents in the dish by adding the reagents_to_add list is exactly what i did and how i got the numbers i put in

You did a manual calculation, I'm saying it should do it dinamically by traversing the list, so when/if the reagents are changed there would be no need to update the formula or risk of inconsistency if someone forgets to

Comment (tomixcomics) [reply code/modules/reagents/reagent_containers/food/snacks/noodles.dm:193]: i see.
still no clue how to do that though. will replacing it with "var/percent = round((reagents.total_volume / volume) * 100)" fix it?

Comment (tomixcomics) [reply code/modules/reagents/reagent_containers/food/snacks/noodles.dm]: marking as resolved as this dish has been fixed according to fluffyghoster's request and suggestions

File: code/modules/reagents/reagent_containers/food/snacks/cultural/unathi.dm (hunk 2 above)
Comment (tomixcomics) [reply code/modules/reagents/reagent_containers/food/snacks/cultural/unathi.dm]: fixed as requested

File: code/modules/reagents/reagent_containers/food/snacks/cultural/unathi.dm (hunk 3 above)
Comment (tomixcomics) [reply code/modules/reagents/reagent_containers/food/snacks/cultural/unathi.dm]: marking as resolved, applied suggested fix

**Review by Arrow768 — APPROVED** [review]

File: code/modules/reagents/reagent_containers/food/snacks/cultural/unathi.dm (hunk 4 above)
Comment (tomixcomics) [reply code/modules/reagents/reagent_containers/food/snacks/cultural/unathi.dm:308]: couldn't find better way to phrase the code, checked with maintainer, they said as long as the code is used in an existing dish it should be fine, marking this as resolved

**Comment by tomixcomics** [comment]
made relevant fixes, marked as resolved one that doesn't need to be changed, ty to all who reviewed, awaiting merge now :)

**Review by NonQueueingMatt — APPROVED** [review]

**Pull request merged**